addons/binding/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/discovery/AstroDiscoveryService.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.astro.discovery;
+
+import static org.openhab.binding.astro.AstroBindingConstants.*;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.model.script.actions.HTTP;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * The {@link AstroDiscoveryService} tries to automatically discover the geolocation based on the internet IP address.
+ *
+ * @author Gerhard Riegler
+ */
+public class AstroDiscoveryService extends AbstractDiscoveryService {
+    private static final Logger logger = LoggerFactory.getLogger(AstroDiscoveryService.class);
+    private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+
+    /**
+     * Creates a AstroDiscoveryService with disabled autostart.
+     */
+    public AstroDiscoveryService() {
+        super(ImmutableSet.of(new ThingTypeUID(BINDING_ID, "-")), DISCOVER_TIMEOUT_SECONDS, false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void startScan() {
+        logger.debug("Starting Astro discovery scan");
+        String result = HTTP.sendHttpGetRequest("http://ip-api.com/json/?fields=lat,lon");
```

Comment:
Please use HttpUtil class directly - a binding must not have any dependency to a "model" bundle.